### PR TITLE
Microsoft.ContainerService: clarify BYO Kubelet identity API

### PR DESCRIPTION
**Pull request by stevekuznetsov**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

**Review by haitch — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json [hunk 1]
```diff
@@ -4982,7 +4982,7 @@
           "additionalProperties": {
             "$ref": "#/definitions/UserAssignedIdentity"
           },
-          "description": "Identities associated with the cluster."
+          "description": "The user identity associated with the managed cluster. This identity will be used in the kubelet. Only one user assigned identity is allowed, pass {\"kubeletIdentity\":{\"resourceId\":\"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}\"}}.",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json:4985]: Might be clearer for other contexts where raw JSON isn't the input to phrase it something like:
```suggestion
          "description": "The user identity associated with the managed cluster. This identity will be used by the kubelet. Only one user assigned identity is allowed. Only only accepted key is \"kubeletidentity\", with value of \"resourceId\": \"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}\".",
```

**Review by matthchr — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json (hunk 1 above)
Comment (stevekuznetsov) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json]: Done!

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json [hunk 2]
```diff
@@ -4982,7 +4982,7 @@
           "additionalProperties": {
             "$ref": "#/definitions/UserAssignedIdentity"
           },
-          "description": "Identities associated with the cluster."
+          "description": "The user identity associated with the managed cluster. This identity will be used by the kubelet. Only one user assigned identity is allowed. Only only accepted key is \"kubeletidentity\", with value of \"resourceId\": \"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}\".",
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json:4985]:
```suggestion
          "description": "The user identity associated with the managed cluster. This identity will be used by the kubelet. Only one user assigned identity is allowed. The only accepted key is \"kubeletidentity\", with value of \"resourceId\": \"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}\".",
```

**Review by matthchr — APPROVED** [review]

**Review by matthchr — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json [hunk 3]
```diff
@@ -4982,7 +4982,7 @@
           "additionalProperties": {
             "$ref": "#/definitions/UserAssignedIdentity"
           },
-          "description": "Identities associated with the cluster."
+          "description": "The user identity associated with the managed cluster. This identity will be used by the kubelet. Only one user assigned identity is allowed. The only accepted key is \"kubeletidentity\", with value of \"resourceId\": \"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}\".",
```
Comment (FumingZhang) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2024-09-01/managedClusters.json:4985]:
```suggestion
          "description": "The user identity associated with the managed cluster. This identity will be used by the kubelet. Only one user assigned identity is allowed. The only accepted key is \"kubeletidentity\", with value of \"resourceId\": \"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}\"."
```

**Comment by FumingZhang** [comment]
Need to fix the failed [Swagger SpellCheck ](https://github.com/Azure/azure-rest-api-specs/pull/30958/checks?check_run_id=31388243984), could add the unknown word `kubeletidentity` to words under category `"filename": "**/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/**/*json"` in file `cSpell.json` to bypass the validation.

**Comment by FumingZhang** [comment]
BTW, I've applied the change to preview API in PR #30976

**Review by matthchr — APPROVED** [review]

**Pull request merged**